pkg/vsphere/tasks/waiter.go

```diff
@@ -70,10 +75,18 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 		}
 
 		log.Errorf("task failed: %s", err)
-		if !isTaskInProgress(err) {
+		if !needsRetry(target, err) {
 			return info, err
 		}
 
+		if needsFix(target, err) {
+			if nerr := fixTask(ctx, sess, target); nerr != nil {
+				log.Errorf("Failed to fix task failure: %s", nerr)
```

Comment:
this method is not retry, it's only fix the object. task will retry in existing way